src/py/flwr/common/address.py

```diff
@@ -22,14 +22,14 @@
 IPV6: int = 6
 
 DOMAIN_PATTERN: Pattern[str] = re.compile(
-    r"^(localhost)|(([a-zA-Z]{1})|([a-zA-Z]{1}[a-zA-Z]{1})|"
+    r"^(https?:\/\/)?(localhost)|(([a-zA-Z]{1})|([a-zA-Z]{1}[a-zA-Z]{1})|"
     r"([a-zA-Z]{1}[0-9]{1})|([0-9]{1}[a-zA-Z]{1})|"
```

Comment:
Whats this going to match? Do you have an example? Are you sure about the final `|`

Reply:
Most of this regex is from https://validators.readthedocs.io/en/latest/_modules/validators/domain.html#domain, we just added `localhost`, `http://`, and `https://` to it.

Reply:
To answer your question more specifically, I think the requirements it will match are the following:

- A domain name consists of minimum two and maximum 63 characters.
- All letters from a to z, all numbers from 0 to 9 and a hyphen (-) are possible.
- A domain name mustn't consist of a hyphen (-) on the third and fourth position at the same time

You can see some basic examples of this in the `address_test.py`.